docs/reddit-migration/considerations/sorting.md

```diff
@@ -0,0 +1,34 @@
+### Sorting
+
+There are several areas where reddit sorts objects coming back from their APIs in a way that isn't necessarily trivial via standard django ORM methods. This list is meant to be comprehensive so we know the problem space:
+
+- `Hot` (named "Active" in our UI)
+  - A logarithmic sort, where as an item ages, it requires an order of magnitude more `upvotes - downvotes` to be sorted above a newer item, the interval for the order of magnitude increase is 12.5 hours
+  - Implementation: https://github.com/mitodl/reddit/blob/stable/r2/r2/lib/db/_sorts.pyx#L47
+- `Confidence` (named "Best" in our UI)
+  - Items are sorted with a [weighted algorithm](https://www.evanmiller.org/how-not-to-sort-by-average-rating.html)
+  - See the corresponding code in the reddit source:
+    - https://github.com/mitodl/reddit/blob/stable/r2/r2/lib/db/_sorts.pyx#L70
+    - https://github.com/mitodl/reddit/blob/stable/install/setup_postgres.sh#L42
+- `New` - posts are sorted in reverse chronological order
+- `old` - posts are sorted in chronological order
+
+
+#### Posts
+
+Posts are sorted in two places: channels and the frontpage. These both support seemingly similar sorts but the implementations actually differ.
+
+##### Posts in Channels
+
+Post sorting in channels is the more straightforward version of post sorting, there are 3 kinds we use: `Hot`, `New`, `Top`
+
+##### Posts in Frontpage
+
+ - `Hot` - posts are sorted using a normalized version of the hot algorithm. This is done to ensure that content from less active channels isn't overshadowed by content from more popular channels.
+   - See the corresponding code in the reddit source:
+     - https://github.com/mitodl/reddit/blob/stable/r2/r2/lib/normalized_hot.py#L38-L79
+ - `New` and `Top` - same method as channels
+
+ #### Comments
+
+ Comments only have a few sorting options we're currently using: `New`, `Old`, and `Confidence`
```

Comment:
does it make sense to think about storing these sorts in a materialized view or something so we're not always recomputing? I guess the frontpage (which is the thing I was thinking of, bc of the email send) is different for every users anyhow so we wouldn't get much mileage out of something like that there. Maybe we could go ahead and try to implement our own sorting and then see how it shakes out performance wise and see if we have any need of something like that.

Reply:
I don't think a materialized view on top of treebeard has much benefit in most cases, but we can always look into it as an option if we're not liking the out-of-the-box performance.

Reply:
oh sorry! meant for that comment to be w/r/t to the whole file, and I was mainly thinking about posts. but yeah we probably don't need to worry about that too much.